tree/dataframe/test/dataframe_simple.cxx

```diff
@@ -513,6 +514,113 @@ TEST_P(RDFSimpleTests, BookCustomAction)
    EXPECT_EQ(*maxSlot, nWorkers-1);
 }
 
+class StdDevHelper {
+private:
+   std::default_random_engine fGenerator;
+   std::normal_distribution<double> fDistribution;
+
+public:
+   std::vector<double> GenerateNumbers(int n)
+   {
+      std::vector<double> numbers;
+      for (int i = 0; i < n; ++i)
+         numbers.push_back(fDistribution(fGenerator));
+      return numbers;
+   }
+
+   double stdDevFromDefinition(const std::vector<double> &samples)
+   {
+      // Calculating the Variance using the definition
+      int nSamples = samples.size();
+      double mean = 0;
+      for (int i = 0; i < nSamples; ++i) {
+         mean += samples[i];
+      }
+      mean = mean / nSamples;
+
+      double varianceRight = 0;
+
+      for (int i = 0; i < nSamples; ++i) {
+         varianceRight += std::pow((samples[i] - mean), 2);
+      }
+      varianceRight = varianceRight / (nSamples - 1);
+      return std::sqrt(varianceRight);
+   }
+
+   double stdDevFromWelford(const std::vector<double> &samples)
+   {
+      ROOT::RDataFrame d(samples.size());
+      return *d.DefineSlotEntry("x",
+                                [&samples](unsigned int slot, ULong64_t entry) {
+                                   (void)slot;
+                                   return samples[entry];
+                                })
+                 .StdDev("x");
+   }
+};
+
+TEST(RDFSimpleTests, StandardDeviation)
+{
+   RDataFrame rd1(8);
+   auto stdDev = rd1.StdDev<ULong64_t>("tdfentry_");
+   EXPECT_NEAR(*stdDev, 2.4494897427832, 0.0000000000001);
+}
+
+TEST(RDFSimpleTests, StandardDeviationPrecision)
+{
+   const int maxNSamples = 100;
+   const int step = 10;
+   const int nTrials = 1;
+
+   std::vector<double> samples;
+   StdDevHelper helper;
+
+   for (int j = 2; j < maxNSamples; j += step) {
+      for (int i = 0; i < nTrials; ++i) {
+         samples = helper.GenerateNumbers(j);
+         auto varianceFromDef = helper.stdDevFromDefinition(samples);
+         auto varianceFromWel = helper.stdDevFromWelford(samples);
+         EXPECT_NEAR(varianceFromDef, varianceFromWel, 0.0000000000001);
+      }
+   }
+}
+
+TEST(RDFSimpleTests, StandardDeviationCollections)
+{
+   RDataFrame tdf(3);
+   auto stdDev = tdf.Define("vector",
+                            []() {
+                               std::vector<int> v(3);
+                               v[0] = 0;
+                               v[1] = 1;
+                               v[2] = 2;
+                               return v;
+                            })
+                    .StdDev<std::vector<int>>("vector");
+   EXPECT_NEAR(*stdDev, 0.86602540378444, 0.0000000000001);
```

Comment:
this _really_ can't be `EXPECT_DOUBLE_EQ`?

Reply:
If I try it, I get this:
```
128: tree/dataframe/test/dataframe_simple.cxx:567: Failure
128:       Expected: *stdDev
128:       Which is: 2.4494897427831779
128: To be equal to: 2.4494897427832
128:       Which is: 2.4494897427832001
```